roles/branch-validation/tasks/main.yml

```diff
@@ -71,6 +71,28 @@
     wait: yes
   with_items: groups[instance_group]
   register: instances
+  when: not groups['compute'] | intersect(groups['controller']) | length | bool
```

Comment:
This role is running on `controller[0]`, so we already know we're in the controller group. Thus all we should have to check for is if we're ALSO in the compute group, so you could do `when: not 'compute' in groups`.

However, I'm even more unclear what the approach is here, why there are two different instance launching tasks. Why can't they be combined into one and just iterate over the hosts in groups['compute'] ?